### PR TITLE
Added `subscribeToMore` docs to `api.md`

### PR DESCRIPTION
**Pull request by alexanderson1993**
Includes arguments and an example code snippet.

### CODE REVIEW

File: source/api.md [hunk 1]
```diff
@@ -500,7 +500,42 @@ data.fetchMore({
...
+**Example:**
+
+```js
+class SubscriptionComponent extends Component {
+  constructor(props){
+    super(props);
+    this.subscription = null;
+    ...
+  }
+    componentWillReceiveProps(nextProps) {
```
Comment (stubailo) [inline source/api.md]: The indentation here is weird.

File: source/api.md [hunk 2]
```diff
@@ -500,7 +500,42 @@ data.fetchMore({
...
+    if (!this.subscription && !nextProps.data.loading) {
```
Comment (stubailo) [inline source/api.md]: I'd add something to stop the previous subscription when props are changed.

**Comment by alexanderson1993** [comment]
I wrote the description and example based on what I'm used to. Let me know if I'm missing anything.

File: source/api.md [hunk 3]
```diff
@@ -500,7 +500,42 @@ data.fetchMore({
...
+
+- `[document]`: Document is a required property that accepts a GraphQL subscription created with `graphql-tag`’s `gql` template string tag. It should contain a single GraphQL subscription operation with the data that will be returned.
+- `[variables]`: The optional variables you may provide that will be used with the `document` option.
+- `[updateQuery]`: A required function run every time the server sends an update. This modifies the results of the HOC query. The first argument, `previousResult`, will be the previous data returned by the query you defined in your `graphql()` function. The second argument is an object with two properties. `subscriptionData` is result of the subscription. `variables` is the variables object used with the subscription query. Using these arguments you should return a new data object with the same shape as the GraphQL query you defined in your `graphql()` function.
```
Comment (stubailo) [inline source/api.md]: I'd mention that this is just like the callback for `fetchMore`, which is an API people might be more familiar with.

**Comment by stubailo** [comment]
Looks great, thanks!

**Pull request merged**